### PR TITLE
PDA5 Kie User Authentication

### PR DESCRIPTION
**Pull request by ffleandro**
Send authenticated user from keycloak to red hat pam server.

### CODE REVIEW

File: src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java [hunk 1]
```diff
@@ -153,10 +162,15 @@ private KieTaskDetails getTaskDetails(RestTemplate restTemplate, Connection conn
         }
     }
 
-    private String createFilters(PagedListRequest request) {
+    public String createUserFilter(Connection connection, AuthenticatedUser user) {
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java:165]: Access can be private.

File: src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java [hunk 2]
```diff
@@ -153,10 +162,15 @@ private KieTaskDetails getTaskDetails(RestTemplate restTemplate, Connection conn
...
+        String username = user == null ? connection.getUsername() : user.getAccessToken().getPreferredUsername();
+        return "?user=" + username;
+    }
+
+    public String createFilters(PagedListRequest request) {
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java:170]: Access can be private.

**Review by werbth — COMMENTED** [review]
@ffleandro Could you include `<includeTests>true</includeTests>` to PMD configuration and fix the issues that arise on the test classes?

File: src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java (hunk 1 above)
Comment (ffleandro) [reply src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java:165]: These can be useds in Unit tests.

File: src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java (hunk 2 above)
Comment (ffleandro) [reply src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java:170]: These can be used in Unit tests

**Comment by werbth** [comment]
test this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by werbth** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Comment by werbth** [comment]
PMD rules for unit test code will be handled in another PR.

**Pull request merged**